controllers/capabilities/developeraccount_threescale_reconciler.go

```diff
@@ -0,0 +1,235 @@
+		params["monthly_charging_enabled"] = strconv.FormatBool(*s.resource.Spec.MonthlyChargingEnabled)
+	}
+
+	return s.threescaleAPIClient.Signup(params)
+}
+
+func (s *DeveloperAccountThreescaleReconciler) findDeveloperAdminUserCR() (*capabilitiesv1beta1.DeveloperUser, error) {
+	queryOpts := []client.ListOption{
+		// query filter by namespace
+		client.InNamespace(s.resource.Namespace),
+	}
+
+	// NICE TO HAVE: use client.MatchingFields in queryOpts. Requires declaring index field in the manager
+	// https://book.kubebuilder.io/cronjob-tutorial/controller-implementation.html#setup
+	// Filter by parent account
+	parentAccountFilter := func(developerUser *capabilitiesv1beta1.DeveloperUser) (bool, error) {
+		return developerUser.Spec.DeveloperAccountRef.Name == s.resource.Name, nil
+	}
+
+	// NICE TO HAVE: use client.MatchingFields in queryOpts. Requires declaring index field in the manager
+	// https://book.kubebuilder.io/cronjob-tutorial/controller-implementation.html#setup
+	// Filter by role
+	adminRoleFilter := func(developerUser *capabilitiesv1beta1.DeveloperUser) (bool, error) {
+		return developerUser.Spec.Role != nil && *developerUser.Spec.Role == "admin", nil
+	}
+
+	// Filter by orphan condition
+	// the account create operation also creates the developer user,
+	// so the search result must include only orphan objects
+	orphanFilter := func(developerUser *capabilitiesv1beta1.DeveloperUser) (bool, error) {
+		return developerUser.IsOrphan(), nil
+	}
+
+	devUserList, err := controllerhelper.FindDeveloperUserList(s.logger, s.Client(), queryOpts,
+		parentAccountFilter,
+		adminRoleFilter,
+		// Filter by providerAccount
+		controllerhelper.DeveloperUserProviderAccountFilter(s.Client(), s.resource.Namespace, s.providerAccountHost, s.logger),
+		orphanFilter,
+	)
+
+	if err != nil {
+		return nil, err
+	}
+
+	// take the first one, anyone should be valid.
+	if len(devUserList) > 0 {
+		return &devUserList[0], nil
+	}
+
+	return nil, nil
+}
+
+func (s *DeveloperAccountThreescaleReconciler) syncDeveloperAccount(devAccount *threescaleapi.DeveloperAccount) (*threescaleapi.DeveloperAccount, error) {
+	update := false
+	deltaAccount := &threescaleapi.DeveloperAccount{
+		Element: threescaleapi.DeveloperAccountItem{
+			ID: devAccount.Element.ID,
+		},
+	}
+
+	if devAccount.Element.OrgName == nil || *devAccount.Element.OrgName != s.resource.Spec.OrgName {
+		update = true
+		deltaAccount.Element.OrgName = &s.resource.Spec.OrgName
+	}
+
+	// MonthlyBilling defaults to True
+	desiredMonthlyBillingEnabled := true
+	if s.resource.Spec.MonthlyBillingEnabled != nil {
+		desiredMonthlyBillingEnabled = *s.resource.Spec.MonthlyBillingEnabled
+	}
+	if devAccount.Element.MonthlyBillingEnabled != nil && *devAccount.Element.MonthlyBillingEnabled != desiredMonthlyBillingEnabled {
+		update = true
+		deltaAccount.Element.MonthlyBillingEnabled = &desiredMonthlyBillingEnabled
+	}
+
+	// MonthlyChargingEnabled defaults to False
```

Comment:
Is the comment outdated or is the value incorrect?

Reply:
fixed